playbooks/open-ondemand.yml

```diff
@@ -0,0 +1,383 @@
+      blockinfile:
+        path: "{{ ood_codeserver_app_dir }}/form.yml"
+        block: |
+          {{ ood_codeserver_app_form }}
+        create: yes
+      tags:
+        - configure
+    - name: configure vs code app manifest
+      blockinfile:
+        path: "{{ ood_codeserver_app_dir }}/manifest.yml"
+        block: |
+          {{ ood_codeserver_app_manifest }}
+        create: yes
+      tags:
+        - configure
+    - name: configure vs code app submit
+      blockinfile:
+        path: "{{ ood_codeserver_app_dir }}/submit.yml.erb"
+        block: |
+          {{ ood_codeserver_app_submit }}
+        create: yes
+      tags:
+        - configure
+    - name: configure vs code app script
+      blockinfile:
+        path: "{{ ood_codeserver_app_dir }}/template/script.sh.erb"
+        block: |
+          {{ ood_codeserver_app_script }}
+        create: yes
+        mode: 0755
+      tags:
+        - configure
+
+- hosts: slurm-node
+  become: yes
+  tasks:
+    - name: install turbovnc
+      apt:
+        deb: https://downloads.sourceforge.net/project/turbovnc/2.2.3/turbovnc_2.2.3_amd64.deb
+      tags:
+        - vnc
+    - name: install codeserver
+      unarchive:
+        src: https://github.com/cdr/code-server/releases/download/2.1698/code-server2.1698-vsc1.41.1-linux-x86_64.tar.gz
+        remote_src: yes
+        dest: /usr/local/bin
+        exclude:
+          - README.md
+          - LICENSE.txt
+          - ThirdPartyNotices.txt
+        creates: /usr/local/bin/code-server
+        mode: 0755
+        extra_opts:
+          - --strip-components=1
+      tags:
+        - codeserver
+
+# Set up LinuxHost adapter 
+- hosts: slurm-master
+  become: yes
+  vars:
+    cluster_config_dir: /etc/ood/config/clusters.d
+    singularity_image_dir: /mnt/shared/singularity-images
+    ood_linuxhost_adapter_script: |
+      #!/bin/bash
+      set -e
+
+      PAM_UID=$(id -u "${PAM_USER}")
+
+      if [ "${PAM_SERVICE}" = "sshd" -a "${PAM_UID}" -ge 1000 ]; then
+              /usr/bin/systemctl set-property "user-${PAM_UID}.slice" \
+                      MemoryAccounting=true MemoryLimit=64G \
+                      CPUAccounting=true \
+                      CPUQuota=700%
+      fi
+    ood_linuxhost_adapter_config: |
+      ---
+      v2:
+        metadata:
+          title: "NVIDIA DeepOps Cluster"
+        login:
+          host: "localhost"
+        job:
+          adapter: "linux_host"
+          submit_host: "localhost"
+          ssh_hosts:
+            - localhost
+          site_timeout: 7200
+          debug: true
+          singularity_bin: /usr/bin/singularity
+          singularity_bindpath: /etc,/media,/mnt,/opt,/run,/srv,/usr,/var,/users
+          singularity_image: {{ singularity_image_dir }}/ubuntu-18.04.simg
+          # Enabling strict host checking may cause the adapter to fail if the user's known_hosts does not have all the roundrobin hosts
+          strict_host_checking: false
```

Comment:
This should probably go in a vars file rather than inline in the playbook. It (mostly) doesn't look like configuration, but consider putting a file in a path like `playbooks/vars/open-ondemand.yml`?